.github/workflows/x-keycloak-operator-hub-publish.yml

```diff
@@ -0,0 +1,155 @@
+name: X Keycloak Operator Hub publish
+
+on:
+  workflow_call:
+    inputs:
+      gh-org:
+        required: true
+        type: string
+      quay-org:
+        required: true
+        type: string
+      community-operators-repo:
+        required: true
+        type: string
+      prod-operators-repo:
+        required: true
+        type: string
+      mvn-url:
+        required: true
+        type: string
+      version:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+    secrets:
+      GH_TOKEN:
+        required: true
+      MVN_USERNAME:
+        required: true
+      MVN_TOKEN:
+        required: true
+      QUAY_USERNAME:
+        required: true
+      QUAY_TOKEN:
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup Java
+        uses: actions/setup-java@v3.0.0
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: Checkout Repository
+        uses: actions/checkout@v3.0.0
+        with:
+          repository: '${{ inputs.gh-org }}/keycloak'
+          token: ${{ secrets.GH_TOKEN }}
+          ref: ${{ inputs.tag }}
+          path: keycloak
+
+      - name: Login to Quay
+        uses: docker/login-action@v1.14.1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Build
+        env:
+          MAVEN_ID: kc-rel-repository
+          MAVEN_URL: ${{ inputs.mvn-url }}
+          MAVEN_USERNAME: ${{ secrets.MVN_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.MVN_TOKEN }}
+        working-directory: keycloak
+        # hardcoded Quay.io image coordinates here are used only for building the K8s YAML resources (where we use Quay and not Docker Hub)
+        run: |
+          ./mvnw clean package \
+              -s ./.github/mvn-rel-settings.xml \
+              -f operator/pom.xml \
+              -DskipTests \
+              -Dquarkus.container-image.image=quay.io/${{ inputs.quay-org }}/keycloak-operator:${{ inputs.tag }} \
+              -Dquarkus.container-image.build=true \
+              -Dquarkus.jib.image-id-file=image-id.txt \
+              -Doperator.keycloak.image=quay.io/${{ inputs.quay-org }}/keycloak:${{ inputs.tag }}
+
+      - name: Install Yq
+        working-directory: keycloak
+        run: sudo snap install yq
+
+      - name: Install OLM
+        working-directory: keycloak
+        run: ./operator/scripts/install-olm.sh
+
+      - name: Fetch previous version of Keycloak
+        working-directory: keycloak
+        # Could be used version of the last tag from the keycloak-bot/community-operators repo
+        run: echo ::set-output name=previous_tag::$(git describe --tags $(git rev-list --tags --max-count=1))
```

Comment:
@stianst I guess this is ok, right? We will never be releasing a micro for prev major after the next major. E.g. we won't do something like 20.0.0 → 21.0.0 → 20.0.1, right?

@mabartos Additionally, please see my next comment around `inputs.version`.

Reply:
Hmmm, thinking about it more, I think maybe for now we should not calculate the previous version as it might cause problems e.g. when re-running some failed flow. Instead, we might want this as an additional parameter. WDYT @mabartos @stianst ?

Reply:
@vmuzikar TBH, I'm not sure if it's better to manually specify the version instead of the used approach. We're trying to have the process "fully" automated, and this might be a step back. This solution is "tag-based" and implies some sort of confidence in the functionality as these tags are immutable and will be continuously chronological, right? 

Could you please elaborate a little bit more your concerns? Thank you